packages/eslint-plugin-template/src/rules/interactive-supports-focus.ts

```diff
@@ -0,0 +1,86 @@
+import type { TmplAstElement } from '@angular-eslint/bundled-angular-compiler';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+import { getDomElements } from '../utils/get-dom-elements';
+import { isHiddenFromScreenReader } from '../utils/is-hidden-from-screen-reader';
+import {
+  isInteractiveElement,
+  isNonInteractiveRole,
+} from '../utils/is-interactive-element';
+import { isDisabledElement } from '../utils/is-disabled-element';
+import { isPresentationRole } from '../utils/is-presentation-role';
+
+type Options = [];
+export type MessageIds = 'interactiveSupportsFocus';
+export const RULE_NAME = 'interactive-supports-focus';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that elements with interactive handlers like `(click)` are focusable.',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      interactiveSupportsFocus:
+        'Elements with interaction handlers must be focusable.',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      Element$1(node: TmplAstElement) {
+        const elementType = node.name;
+        if (!getDomElements().has(elementType)) {
+          return;
+        }
+
+        const tabIndex = [...node.attributes, ...node.inputs].find(
```

Comment:
It looks like `tabIndex` is not used until the second conditional. Let's move it to line 66?